data_access_service/core/routes.py

```diff
@@ -214,96 +226,127 @@ def _remove_file_if_exists(file_path):
     return response
 
 
-@restapi.route("/health", methods=["GET"])
-def health_check() -> Response:
-    return Response("healthy", mimetype="application/json")
+class HealthCheckResponse(BaseModel):
+    status: str
+    status_code: int
+
+
+def get_api_instance(request: Request) -> API:
+    instance = request.app.state.api_instance
+    if not instance.is_ready:
+        raise HTTPException(
```

Comment:
raise Exception here but then health_check do not have try catch? Is it necessary to raise Exception?